flang/lib/Semantics/canonicalize-omp.cpp

```diff
@@ -131,20 +134,59 @@ class CanonicalizationOfOmp {
       // Ignore compiler directives.
       if (GetConstructIf<parser::CompilerDirective>(*nextIt))
         continue;
+      // Keep track of the loops to handle the end loop directives
+      llvm::SmallVector<parser::OpenMPLoopConstruct *> loops;
+      loops.push_back(&x);
+      if (auto *innerConstruct{
```

Comment:
```suggestion
      if (auto *innerConstruct{GetOmpIf<parser::OpenMPLoopConstruct>(*nextIt)}) {
```

We can simplify here and get the OpenMPLoopConstruct directly.